apis/v1alpha1/secret.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SecretSpec defines the desired state of Secret.
+type SecretSpec struct {
+
+	// A list of Regions and KMS keys to replicate secrets.
+	AddReplicaRegions []*ReplicaRegionType `json:"addReplicaRegions,omitempty"`
```

Comment:
This field isn't well named for a CRD. It comes from the `CreateSecretInput.AddReplicaRegions` shape. Ideally this would be called `ReplicaRegions`, that way it represents the desired state of replica regions during creation and update.

Reply:
rename [rename AddReplicaRegions to ReplicaRegions](https://github.com/aws-controllers-k8s/secretsmanager-controller/pull/2/commits/db2d728ded905430de63b85a601a028cbced74cd)